xdsl/transforms/eqsat_serialize_egraph.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+from xdsl.context import Context
+from xdsl.dialects import builtin, eqsat, func
+from xdsl.ir import BlockArgument, Operation
+from xdsl.passes import ModulePass
+
+
+def serialize_func_to_egraph(f_op: func.FuncOp):
```

Comment:
Could you walk the whole IR backwards, instead of just per function? Should give the same results, and be more generic.

Reply:
You mean, do the walk over the module and handle every FuncOp like I did here, or something else?
(also, why reverse?)

Reply:
The reason I only considered funcs is that I don't think e.g. scf.for bodies should be introspected and serialized.

Reply:
Backwards because then you only need to process each operation once, and I don't think that this is the right place to put logic to restrict what is introspected, we might as well blindly trust the use-def structure.